hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-tests/src/test/java/org/apache/hadoop/yarn/server/MiniYARNCluster.java

```diff
@@ -456,6 +456,19 @@ public int getActiveRMIndex() {
     return -1;
   }
 
+  /**
+   * Stopping MiniYarnCluster can take more than 30 seconds and it's a blocking operation so it blocks test execution,
+   * so to prevent that we can stop the cluster asynchronously,
+   * that can ultimately save test execution time.
+   * @param owner the Test class that owns the instance of MiniYarnCluster
+   */
+  public void asyncStop(Object owner) {
```

Comment:
Not sure how future proof is this, I mean 100 small test functions where each complete in <100ms and 100 thread will be created for stopping each mini cluster.

Also the parameter owner is basically just an additional context (string) for the thread name, it's kind of misleading.

I think we could leave this as a sync operation for these two test methods.

Reply:
@susheelgupta7  Thanks for the review fixes. I think we should address this too. Just simply stop the yarn mini cluster synchronously.

Reply:
Each mini cluster is taking 40-55 secs to be stopped. As per your comment and the code, async  is an inappropriate choice so updated with sync stopping of Mini cluster.
 Thanks @tomicooler for reviewing.